modules/dnn/test/test_backends.cpp

```diff
@@ -70,14 +70,15 @@ class DNNTestNetwork : public TestWithParam <tuple<DNNBackend, DNNTarget> >
         netDefault.setInput(inp);
         Mat outDefault = netDefault.forward(outputLayer).clone();
 
-        net.setInput(inp);
         net.setPreferableBackend(backend);
         net.setPreferableTarget(target);
         if (backend == DNN_BACKEND_HALIDE && !halideScheduler.empty())
         {
             halideScheduler = findDataFile(halideScheduler, false);
             net.setHalideScheduler(halideScheduler);
         }
+
+        net.setInput(inp);
```

Comment:
Can we convert input blob to FP16 at the network's initialization stage (i.e. setUpNet)? Before the first `forward` call we could call any `net.set*` methods in any order.

Reply:
code is updated, net.set* can be in any order before forward call.